### ISSUE TITLE
Cant parse <tr> without <table> element.

### ISSUE THREAD

**Issue by the-hotmann** [PuerkitoBio/goquery#458]
I have a table, which does have many rows (`<tr>`):

```
<table>
    <tr class="text-nowrap">
        <td class="bg-success-subtle">TEXT1</td>
        <td class="bg-primary-subtle">TEXT2</td>
        <td class="bg-primary-subtle">TEXT3</td>
        <td class="bg-warning-subtle">TEXT4</td>
        <td class="bg-warning-subtle">TEXT5</td>
    </tr>
    <tr class="text-nowrap">
        <td class="bg-success-subtle">TEXT1</td>
        <td class="bg-primary-subtle">TEXT2</td>
        <td class="bg-primary-subtle">TEXT3</td>
        <td class="bg-warning-subtle">TEXT4</td>
        <td class="bg-warning-subtle">TEXT5</td>
    </tr>
</table>
```

So I store them all in an array of variables to sort them.

```
var trElements []string
```

Now there is `trElements[0]` and `trElements[1]` which I want to parse.
I want to get the first Text of the first `<td>` element.  To simulate this, I have prepared this demo:

```
package main

import (
	"fmt"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

func main() {
	html := `<tr class="text-nowrap">
        <td class="bg-success-subtle">TEXT1</td>
        <td class="bg-primary-subtle">TEXT2</td>
        <td class="bg-primary-subtle">TEXT3</td>
        <td class="bg-warning-subtle">TEXT4</td>
        <td class="bg-warning-subtle">TEXT5</td>
</tr>`

	doc, err := goquery.NewDocumentFromReader(strings.NewReader(html))
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	test1 := doc.Find("td.bg-success-subtle").Text()
	fmt.Println("Test1:", test1) // Should output: TEXT1

	test2 := doc.Find("td").First().Text()
	fmt.Println("Test2:", test2) // Should output: TEXT1

	test3, _ := doc.Find("td.bg-success-subtle").Html()
	fmt.Println("Test3:", test3) // Should output: TEXT1

	test4, _ := doc.Find("td").First().Html()
	fmt.Println("Test4:", test4) // Should output: TEXT1

	test5 := doc.Find(".bg-success-subtle").Text()
	fmt.Println("Test5:", test5) // Should output: TEXT1

	test6 := doc.Find("tr td").First().Text()
	fmt.Println("Test6:", test6) // Should output: TEXT1

	test7, _ := doc.Find("tr td").First().Html()
	fmt.Println("Test7:", test7) // Should output: TEXT1
}
```

output:
```
Test1: 
Test2: 
Test3: 
Test4: 
Test5: 
Test6: 
Test7: 
```

Which does not work!
As soon as I wrap a `<table>` element around the html it will succeed:

```
package main

import (
	"fmt"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

func main() {
	html := `<table><tr class="text-nowrap">
        <td class="bg-success-subtle">TEXT1</td>
        <td class="bg-primary-subtle">TEXT2</td>
        <td class="bg-primary-subtle">TEXT3</td>
        <td class="bg-warning-subtle">TEXT4</td>
        <td class="bg-warning-subtle">TEXT5</td>
</tr></table>`

	doc, err := goquery.NewDocumentFromReader(strings.NewReader(html))
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	test1 := doc.Find("td.bg-success-subtle").Text()
	fmt.Println("Test1:", test1) // Should output: TEXT1

	test2 := doc.Find("td").First().Text()
	fmt.Println("Test2:", test2) // Should output: TEXT1

	test3, _ := doc.Find("td.bg-success-subtle").Html()
	fmt.Println("Test3:", test3) // Should output: TEXT1

	test4, _ := doc.Find("td").First().Html()
	fmt.Println("Test4:", test4) // Should output: TEXT1

	test5 := doc.Find(".bg-success-subtle").Text()
	fmt.Println("Test5:", test5) // Should output: TEXT1

	test6 := doc.Find("tr td").First().Text()
	fmt.Println("Test6:", test6) // Should output: TEXT1

	test7, _ := doc.Find("tr td").First().Html()
	fmt.Println("Test7:", test7) // Should output: TEXT1
}
```
output:
```
Test1: TEXT1
Test2: TEXT1
Test3: TEXT1
Test4: TEXT1
Test5: TEXT1
Test6: TEXT1
Test7: TEXT1
```

I do not like to have to wrap something around, just to parse it and then remove it again. It would be awesome, if this package could parse even partial table stuff, like I have to deal with :)

Thanks in advance!

**Comment by mna** [comment]
Hello, 

That's the way the html5 parser that goquery uses works. You can look at the generated html once parsed with NewDocumentFromReader (using `goquery.OuterHtml(doc.Selection)`):

```
<html><head></head><body>
        TEXT1
        TEXT2
        TEXT3
        TEXT4
        TEXT5
</body></html>
```

So as you can see there are no `<tr>` tags at all because the document you provided was not valid html, so [the parser follows some rules](https://pkg.go.dev/golang.org/x/net/html#Parse) (the same as the browsers AFAIK) to end up with an html document to work with.

One option is to directly parse with the html5 parser and call `ParseFragment`, providing a `<table>` node as context (https://pkg.go.dev/golang.org/x/net/html#ParseFragment) but at this point this is no different than adding the table in your html and parsing via goquery.

Hope this helps,
Martin

**Comment by the-hotmann** [comment]
You are correct; the HTML provided is not valid.

Initially, I viewed this process as a DOM parser, somewhat similar to an XML parser. However, I now realize that it requires genuine HTML rather than just a portion of it. After I transformed all the `<tr>` elements from the `<table>` into separate elements, I found that I couldn't use the library on them unless, as I did, I wrapped each of them back into a `<table>` element.

This struck me as a bit unusual, but I now understand it. When I inspect a page using developer tools and replace everything with only the <tr> elements, it doesn't display them. So I became perplexed when the HTML I provided didn't seem to be present.

Thank you for clearing up the confusion.

Warm regards,
Martin
